tests/Agent/IntegrationTests/IntegrationTests/RequestHeadersCapture/WCF/AllowAllHeadersDisabledTests.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using System.Linq;
+using MultiFunctionApplicationHelpers;
+using NewRelic.Agent.IntegrationTestHelpers;
+using NewRelic.Agent.IntegrationTestHelpers.Models;
+using NewRelic.Agent.IntegrationTests.Shared.Wcf;
+using Xunit;
+using Xunit.Abstractions;
+using static NewRelic.Agent.IntegrationTests.WCF.WCFTestBase;
+
+namespace NewRelic.Agent.IntegrationTests.RequestHeadersCapture.WCF
+{
+    public abstract class AllowAllHeadersDisabledTests : WCFEmptyTestBase<ConsoleDynamicMethodFixtureFWLatest>
+    {
+        public AllowAllHeadersDisabledTests(ConsoleDynamicMethodFixtureFWLatest fixture, ITestOutputHelper output, HostingModel hostingModel, WCFBindingType bindingType)
+            : base(fixture, output, hostingModel, bindingType) { }
+
+        protected virtual IEnumerable<string> UnallowedHeaders => new[]
+        {
+            "request.headers.cookie",
+            "request.headers.authorization",
+            "request.headers.proxy-authorization",
+            "request.headers.x-forwarded-for"
+        };
+
+        protected virtual IDictionary<string, string> ExpectedHeaders => new Dictionary<string, string>
+        {
+            { "request.headers.referer", "http://example.com/" },
+            { "request.headers.accept", "text/html" },
+            { "request.headers.content-length", GetExpectedContentLength() },
+            { "request.headers.host", $"localhost:{_fixture.RemoteApplication.Port}" },
+            { "request.headers.user-agent", "FakeUserAgent" },
+        };
+
+        protected virtual IEnumerable<string> UnexpectedHeaders => new[]
+        {
+            "request.headers.foo",
+            "request.headers.dashes-are-valid",
+            "request.headers.dashesarevalid"
+        };
+
+        protected string GetExpectedContentLength()
+        {
+            switch (_binding)
+            {
+                case WCFBindingType.BasicHttp: return "168";
```

Comment:
not sure it needs to be changed, as this is a valid pinning test for the existing behavior.. but if anything changes in the binding implementations then this could break in a confusing manner.

Reply:
This definitely causes false-positive test result, and I believe could do better. This can be done as a follow up story.